node/src/git.rs

```diff
@@ -34,6 +34,35 @@ pub enum ListRefsError {
     InvalidRef(#[from] RefError),
 }
 
+pub mod refs {
+    use super::*;
+
+    // Where project information is kept.
+    pub const IDENTITY_BRANCH: &str = "radicle/id";
```

Comment:
You can use [`lazy_static`][0] and the `refname!` macro from `git-ref-format` to define this. We did something similar in `radicle-link`:
```rust
lazy_static! {
    pub static ref DEFAULT_PATH: ext::Qualified = ext::Qualified::from(reflike!("refs/rad/id"));
}
```

[0]: https://docs.rs/lazy_static/1.4.0/lazy_static/

Reply:
Yes, we're using `once_cell` in this crate so it would look like:
```rust
pub static REMOTES_GLOB: Lazy<refspec::PatternString> =
    Lazy::new(|| refspec::pattern!("refs/remotes/*"));
```

Reply:
Why is it so complicated to define a constant in rust?

Reply:
> Why is it so complicated to define a constant in rust?

I _think_ because they need to be primitives that can live on the stack, i.e. known at compile time values. Hence, why you can't use `String` but you can use `&str`. Another way I think about it is that if you need to do a computation to get the constant then it can't be constant.


> Why is this better?

This is an improvement over `&str` because it's a typed representation of `refs/rad/id`, the `Qualified` type guarantees that we a have `refs/` prefix -- as opposed to `&str` being any old value. Here's a great article on the benefits of being more specific about types[[0]].

The `lazy_static`, I believe, works around this computation idea I mentioned above. It will run the computation once and only once, so it's constant after that first run. `once_cell` does something similar.

[0]: https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/

Reply:
@FintanH thanks for the explanation :)

I wasn't aware about how much of it was due to type being used and personally unknown rust features due to other concerns.

Reply:
Yes, worth noting that lazy_static/once_cell do *not* run the computation at compile-time, they simply cache the result after the first run, so it's *kind of* like a constant.